### ISSUE TITLE
Better feedback when entering too large approval code or memo

### ISSUE THREAD

**Issue by kentbower** [kbower/tickettest1#2380]
_Imported from trac ticket 2380.
Created by **sal**
Opened in trac: Jun 18, 2013, 10:55:54 AM
Last modified in trac: Jun 19, 2013, 5:25:10 PM_
***
_by sal:_
It was a debit and the customer had already left with the card because she was late to get back to work. He actually thought the order had saved and then, finally, this error appeared.

[[Image(CaptureXT.png)]]

***
* sjohnson attached [CaptureXT.png](https://trac.retailarchitects.com/trac/attachment/ticket/2380/CaptureXT.png) on Jun 18, 2013, 10:55:54 AM

**Comment by kentbower** [comment]
_**sjohnson** trac comment on Jun 18, 2013, 10:59:32 AM:_

The problem is that they manually put an approval code of '90 DAYS SAC' which is 11 chars but the column can only handle 10. This field is typically not something a human hand enters for most retailers so it looks like we may not be checking the value in there as carefully. If he removes the financing and re-adds it with a shorter approval code, it should work fine.
___
It isn't acceptable to give the user no clue what he did wrong.
Note that the method 'check_column_constraints()' is meant for checking for this problem.

**Comment by kentbower** [comment]
_**kbower** trac comment on Jun 18, 2013, 12:17:32 PM:_

```
'UPDATE orders SET financecompanyid=:financecompanyid, tentativefinancedamount=:tentativefinancedamount, approvalcode=:approvalcode WHERE orders.orderid = :orders_orderid' 
{'financecompanyid': 'LOANT25796', 'approvalcode': 'TOWER LOAN CPO', 'tentativefinancedamount': Decimal('1500'), 'orders_orderid': '0617313TF35'}
```

```
'INSERT INTO artrans (transactionid, customerid, companyid, orderid, adjinvoice, contractid, transactioncodeid, status, paymenttypeid, postdate, writedate, amount, checknumber, cashback, accountnumber, approvalcode, sourcesiteid, source, paymentsiteid, artype, username, cashier, cashdrawer, memo, createdate, sequence, hosttransactionid, nsftransactionid, refundamount, voidamount, voidtransactionid) VALUES (:transactionid, :customerid, :companyid, :orderid, :adjinvoice, :contractid, :transactioncodeid, :status, :paymenttypeid, :postdate, :writedate, :amount, :checknumber, :cashback, :accountnumber, :approvalcode, :sourcesiteid, :source, :paymentsiteid, :artype, :username, :cashier, :cashdrawer, :memo, :createdate, :sequence, :hosttransactionid, :nsftransactionid, :refundamount, :voidamount, :voidtransactionid)' 
{'username': None, 'postdate': datetime.date(2013, 6, 8), 'paymentsiteid': '11', 'companyid': '01', 'checknumber': '21006114028', 'transactioncodeid': 'PMT', 'approvalcode': None, 'orderid': None, 'accountnumber': None, 'voidtransactionid': None, 'sequence': None, 'hosttransactionid': None, 'source': 'APP', 'cashback': None, 'paymenttypeid': 'MO', 'memo': None, 'artype': 'I', 'status': 'T', 'writedate': None, 'adjinvoice': None, 'refundamount': None, 'voidamount': None, 'cashdrawer': '11', 'createdate': datetime.datetime(2013, 6, 8, 16, 17, 41, 979200), 'cashier': 'FJW', 'nsftransactionid': None, 'transactionid': Decimal('12005366'), 'customerid': 'DAVIA83736', 'amount': Decimal('120'), 'contractid': '0313211B46', 'sourcesiteid': '11'}
```

```
'UPDATE artrans SET amount=:amount, memo=:memo, voidamount=:voidamount WHERE artrans.transactionid = :artrans_transactionid' 
{'voidamount': Decimal('100'), 'amount': Decimal('0'), 'memo': 'Card:VI9385 M/Y:08/14 Name:...       (DUPLICATE)', 'artrans_transactionid': Decimal('11971322.0000000000')}
```

**Comment by kentbower** [comment]
_**kbower** trac comment on Jun 19, 2013, 5:25:10 PM:_

(Changeset [[4263]](../commit/cf3335e35fe785611ce15da8ae0f6a3630f705c1))
* fixes #2380, Better feedback when entering too large approval code or memo #all

**Comment by kentbower** [comment]
_**kbower** trac status change on Jun 19, 2013, 5:25:10 PM:_ **closed**
